File: Creational/Lab2/C#/Apartment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace C_
{
    public class Apartment : RealEstate
    {
        public int RoomsNumber{ get; set;}
        public Apartment(string address, string description, double area, double price, string status, int roomsNumber) :
            base(address, description, area, price, status)
        {
            RoomsNumber = roomsNumber;
        }
        public override string ToString()
        {
            return "Apartment:\n" + base.ToString() + $", Rooms number: {RoomsNumber}";
        }

        public override Apartment Clone()
        {
            var clone = (Apartment)MemberwiseClone();
            clone.Realtor = new Realtor(Realtor);
            return clone;
        }

        public override Apartment ShallowCopy()
        {
            return (Apartment)MemberwiseClone();
        }
    }
}

File: Creational/Lab2/Python/apartment.py
from real_estate import *

class Apartment(RealEstate):
    def __init__(self, address, description, area, price, status, rooms_number):
        super().__init__(address, description, area, price, status)
        self.rooms_number = rooms_number

    def __str__(self):
        return f"Apartment:\n{super().__str__()}, Rooms number: {self.rooms_number}"
    
    def clone(self):
        clone = copy.deepcopy(self)
        clone.realtor = Realtor(self.realtor.id, self.realtor.name_and_surname, self.realtor.phone, self.realtor.email)
        return clone

    def shallow_copy(self):
        return copy.copy(self)